gw_eccentricity/load_data.py

```diff
@@ -827,22 +834,24 @@ def load_sxs_catalogformat(**kwargs):
                 "This is required to get the binary parameters which are "
                 "used to eavaluate the zero ecc waveform.")
 
-    # load modes
-    modes_dict = {}
-    data = h5py.File(filepath, "r")
-    waveform_data = data[f"Extrapolated_N{extrap_order}.dir"]
-    for idx, mode in enumerate(mode_array):
-        ell, m = mode
-        mode_data = waveform_data[f"Y_l{ell}_m{m}.dat"]
-        # create the time array only once
-        if idx == 0:
-            time = mode_data[:, 0]
-            t = np.arange(time[0], time[-1], dt)
-        hlm = mode_data[:, 1] + 1j * mode_data[:, 2]
-        amp_interp = interpolate(t, time, np.abs(hlm))
-        phase_interp = interpolate(t, time, -np.unwrap(np.angle(hlm)))
-        hlm_interp = amp_interp * np.exp(-1j * phase_interp)
-        modes_dict.update({(ell, m): hlm_interp})
+    # Check file format
+    if "Strain_N" in kwargs["filepath"]:
+        # find the extrapolation order from file name and check if
+        # it is consistent with the value provided via kwargs.
+        # If they are not consistent, raise warnings and ignore
+        # the value provided in the kwargs.
+        base_name = os.path.basename(kwargs["filepath"])
+        extrap_order_from_filename = int(
+            re.findall(r"\d", base_name)[0])
+        if kwargs["extrap_order"] != extrap_order_from_filename:
+            warnings.warn("Exrapolation order in `kwargs` is "
+                          f"{kwargs['extrap_order']} but waveform file seems "
+                          "to have extrapolation order "
+                          f"{extrap_order_from_filename}. "
+                          "Ignoring the `kwargs` value.")
```

Comment:
Instead of this complicated stuff...don't allow extrap_order in kwargs for the new format. Error is more appropriate here.

Reply:
It becomes trickier. Currently, the idea is that we use the same origin="SXSCatalog" in the load_waveform and depending on the file format, we use either the load_old/load_new. In both cases, they use the same set of default kwargs. But if we want to change the default kwargs, i.e., allow extrap_order in old but not in the new, then we have to change the origin to "SXSCatalog_old" and "SXSCatalog_new" since the default kwargs depend on the origin. Also, in that case, we don't need the common load function but instead just use the separate ones. If we go that route then the user needs to figure out which format it is beforehand and the effort to make a single sxs load function is needed anymore.

Reply:
Ok, but this should still be an exception.
 
But more importantly, you are telling me the code will only work for extrap_order=3 if the user gets it right both in the kwargs and in the filename? If the user does the natural thing: give the file path and because it has the extrap order in it, assume the code is smart enough to figure it out and therefore not specify extrap_order in the kwargs. Then, because it will default to 2 in the kwargs, things will be inconsitent and the user gets a warning/exception? Not nice.

How about the much simpler option of:
Say in the docs for the kwargs that extrap order will be ignored for the new format. Then, just don't check for it.

Reply:
Done